### ISSUE TITLE
known locales list empty after upgrading to 0.18

### ISSUE THREAD

**Issue by tonnenpinguin** [elixir-gettext/gettext#249]
Hi guys,

After updating from 0.17.4 to 0.18 I noticed that `Gettext.known_locales/1` returns an empty list.

Downgrading to 0.17.4 brings everything back to normal and I didn't notice anything in the changelog or diff that would indicate a required change in the config. Any pointers as to what may cause this issue?

The new `allowed_locales` option is not set in my config.

The issue occurs in our persistence module, which does not actually use po files but rather uses Gettext mostly as a storage unit for the users current locale (based on [smeevil/ecto_translate](https://github.com/smeevil/ecto_translate) )

**Comment by josevalim** [comment]
Unfortunately I cannot reproduce it locally using regular Gettext, so please try to provide an application that reproduces the error or a minimal test case. Thanks!

**Comment by tonnenpinguin** [comment]
Thanks for the quick response!

I did some digging and actually figured out the problem!

Presumably because I don't have any actual strings `mix gettext.extract` doesn't actually regenerate the files (which makes sense I guess) and so it used the default files I had in my `priv/gettext` directory.

However, these files where in `priv/gettext/en/default.po` and not in `priv/gettext/en/LC_MESSAGES/default.po` as gettext expected.
After moving the files into the correct directory everything is working as expected again.


If I had to guess this change in behaviour was caused by the changes made to [Gettext.Compiler.known_po_files/2](https://github.com/elixir-gettext/gettext/compare/a8dbcd276a5099bbb3dc473d39c85da48d8093aa...master#diff-f39be9eabead94474491fe6d2d1137abR650) as instead of just looking for the existence of translation directories gettext now actually looks into the files within these directories.
